File: Leetcode/1603 Design Parking System.py
class ParkingSystem:
    def __init__(self, big: int, medium: int, small: int):
        self.cars = [0, big, medium, small]

    def addCar(self, carType: int) -> bool:
        if self.cars[carType] == 0:
            return False
        self.cars[carType] -= 1
        return True

File: Leetcode/1603 Design Parking System.ts
class ParkingSystem {
    private spots: number[];

    constructor(big: number, medium: number, small: number) {
        this.spots = [0, big, medium, small];
    }

    addCar(carType: number): boolean {
        if (this.spots[carType] === 0) {
            return false;
        } 
        this.spots[carType]--;
        return true;
    }
}
